src/shared/UniversalRender.jsx

```diff
@@ -64,15 +69,23 @@ const scrollTop = (el, topOffset, prevDocumentInfo, triesRemaining) => {
     const documentInfo = {
         scrollHeight: document.body.scrollHeight,
         scrollTop: Math.ceil(document.scrollingElement.scrollTop),
-        scrollTarget: calcOffsetRoot(el) + topOffset
+        scrollTarget: calcOffsetRoot(el) + topOffset,
+        direction: prevDocumentInfo.direction,
     };
-
-    if(prevDocumentInfo.scrollTop > (documentInfo.scrollTop + SCROLL_UP_FUDGE_PIXELS)) { //detecting that the user has scrolled in an up direction
-        return;
+    let doScroll = false;
+    if(SCROLL_DIRECTION_DOWN === prevDocumentInfo.direction) {
```

Comment:
a jedi you are not, writing C you are not, Yoda syntax do not use

Reply:
besides, this variable is declared `const` -- it would be a syntax error for it to be reassigned here

Reply:
I like yoga. He's short, green, and snarky. 
FYI, prevDocumentInfo.direction isn't a const, so non-yoda in this expression _is_ / _would be_ vulnerable code.